### ISSUE TITLE
Bug in container width

### ISSUE THREAD

**Issue by Michaelgimii** [elementor/elementor#19758]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

I tried to create a box using a container and display the products inside it : https://prnt.sc/Ughp1XDPr_rg
This container is configured as follows : https://prnt.sc/tPxUuODljEgE

So far no problem. Create a similar container below or above it, again no problem. **But if we give a margin from the top in the lower container, the container will break : **https://prnt.sc/tDvrxdpMa1Ub

### Steps to reproduce

https://user-images.githubusercontent.com/88034902/189241499-e5fed2ec-1c8e-4a7a-8c66-8c93c86ce7a7.mp4

hi @udidol @rami-elementor  ,If you are unable to reproduce according to the video above, please import the following template into Elementor and give the bottom container a margin from the top. The bug is quite obvious.

[elementor-5191-2022-09-08.zip](https://github.com/elementor/elementor/files/9531144/elementor-5191-2022-09-08.zip)

thanks

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

-

**Comment by perks98** [comment]
This is due to having margin auto on the left and right sides of the container, but when a user input for margin is added it resets to a specific value.  Having the ability to choose margin auto has been a feature request for a while now I believe.
![Screenshot_577](https://user-images.githubusercontent.com/55829113/189301372-7372f5f5-7e4f-4a3f-8fcf-f5e570f9cd6e.jpg)

**Comment by Michaelgimii** [comment]
@perks98 
Isn't this a bug anyway? I tested this in bricks builder and it worked fine there. Anyway, I hope there is a solution for it.

**Comment by perks98** [comment]
@Michaelgimii It is just my opinion of course but I think it is more of a feature request as the controls [screenshot below], based on your own video, work as expected as the values are set to 0px. So it isn't necessarily broken just not working as you would have wanted. When the container experiment first came out in beta I think the left and right options were locked like how sections operate but based on user feedback this was removed on all containers so users can have full control over margin, this then leads to your problem.

**Screenshot from your video:** 
![Screenshot_578](https://user-images.githubusercontent.com/55829113/189311831-c3aa504e-0860-4e21-9ce4-ecb8ce39ce60.jpg)

**Existing sections with left and right on 'auto' locked:** 
![Screenshot_579](https://user-images.githubusercontent.com/55829113/189312378-5c51abbf-4031-4196-bded-3179d3663431.jpg)

<hr>

Maybe post a screenshot of how Bricks have implemented their margin calculation and see if the Elementor team can learn and maybe implement a similar feature?

**Comment by Michaelgimii** [comment]
@perks98 , I do not work with BRICS. Just comparing it to Elementor in some cases. I will send a screenshot. Sure

**Comment by Michaelgimii** [comment]
![1](https://user-images.githubusercontent.com/88034902/189320288-bbf29597-8405-44c2-a26b-6d1083772f6d.png)

I think the problem will be solved if elementor works as before only in the main container (picture below you posted).
![189312378-5c51abbf-4031-4196-bded-3179d3663431](https://user-images.githubusercontent.com/88034902/189320606-2754bec5-c6d4-4d10-9e86-7fd94ec9112f.jpg)

Therefore, containers within the parent container do not behave this way. Could this be a useful solution?

**Comment by perks98** [comment]
Based on your Bricks screenshot it looks like the way it differs from Elementor is that you can add input to individual sides [top, right, bottom, left] without making the other values default to 0px upon interaction like it does with Elementor. Perhaps Elementor could look into that, making the default the initial value rather than setting all the sides sizes to 0px when you want to update one field.

**Comment by udidol** [comment]
You are right - the problem here is not the incorrect margin, it's that `full width` mode is not supposed to have a `width` control. That defeats the whole purpose of _"Full Width"_. This control will be removed.

**Comment by Michaelgimii** [comment]
hi @udidol , thanks for the reply.
So here is a limitation in the container?
I wanted to place a container so that it has the same width as the global setting. This container has a shadow box (image below). But no matter how hard I try, this is not possible **with only one container**.

**Comment by udidol** [comment]
Not sure yet. We might leave the control and find a different solution that will allow custom width in full width mode. We will see.

**Comment by Michaelgimii** [comment]
Thnaks again @udidol 
I suggest you check the performance of the bricks builder container. https://try.bricksbuilder.io/
This can be very useful in improving containers in Elementor.

**Comment by udidol** [comment]
Yes, I agree that this is a problem.
The "Width" control in Boxed mode only controls the width of the Container's inner content.
The reason the Full width control was added in containers is that we wanted to let users control the width of the entire container, not just the inner content. This is to allow controlling width that affects things like background (image/slideshow/video etc.).

Since top level containers are centered by default in Elementor, and the centering is done with `margin: auto` - adding any margin to a "Full Width" container with a custom width set to it will remove the default centering of the container.

I understand why this is problematic, but for now, if you want to achieve this result, you can use a container within a full width container.

**Comment by Michaelgimii** [comment]
thank you @udidol
